functests/4_latency/latency.go

```diff
@@ -144,29 +148,38 @@ var _ = Describe("[performance] Latency Test", func() {
 				Skip("no maximum latency value provided, skip buckets latency check")
 			}
 
-			latencies := extractLatencyValues("oslat", `Maximum:\t*([\s\d]*)\(us\)`, workerRTNode)
+			latencies := extractLatencyValues(logName, `Maximum:\t*([\s\d]*)\(us\)`, workerRTNode)
 
 			// under the output of the oslat very often we have one anomaly high value, for example
 			// Maximum:    16543 15 15 14 13 12 12 13 12 12 12 12 12 12 12 12 12 (us)
 			// it still unclear if it oslat bug or the kernel one, but we definitely do not want to
 			// fail our test on it
 			var anomaly bool
-			for _, lat := range strings.Split(latencies, " ") {
+			var anomalyValue int
+			multipleLatencies := false
+			latenciesList := strings.Split(latencies, " ")
+			for _, lat := range latenciesList {
 				if lat == "" {
 					continue
 				}
-
 				curr, err := strconv.Atoi(lat)
 				Expect(err).ToNot(HaveOccurred())
 
-				// skip the anomaly value
-				if curr > maximumLatency && !anomaly {
+				if !anomaly {
```

Comment:
it does not correct, you want to enter the anomaly `if condition` only when `curr > expected_latency` or we can remove this anomaly check at all in the light of https://issues.redhat.com/browse/CNF-1048

Reply:
at this stage we don't apply the check yet, I just save the first value of the latencies list for later, so in case there was only one value we can apply the check on it & not pass the test. however, if the bug is already in absolute, I agree to remove this check.

Reply:
I think you meant obsolete, lets remove these check at all, it will make test clearer

Reply:
yes :)